### ISSUE TITLE
TODO: Add title

### ISSUE THREAD

**Issue by pihentagy** [koalaman/shellcheck#1242]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2059
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

printf $'!!! Too many errors !!!\n'
```

#### Here's what shellcheck currently says:
```
Line 3	SC2059: Don't use variables in the printf format string. Use printf "..%s.." "$foo".
```
#### Here's what I wanted or expected to see:

No errors should be reported

**Comment by koalaman** [comment]
This is a very strange pattern, but still a valid issue. The simpler and more portable way of doing this is `printf '!!! Too many errors !!!\n'`

This happened because ShellCheck didn't correctly see `$'..'` as a literal string, but now it does. Thanks!

**Comment by pihentagy** [comment]
Oh, I didn't know printf accepts escape characters. Good to know.
